homeassistant/components/tfiac/climate.py

```diff
@@ -0,0 +1,190 @@
+"""Climate platform that offers a climate device for the TFIAC protocol."""
+from concurrent.futures import _base as futures
+import logging
+from datetime import timedelta
+
+import voluptuous as vol
+
+from homeassistant.components.climate import PLATFORM_SCHEMA, ClimateDevice
+from homeassistant.components.climate.const import (
+    STATE_AUTO, STATE_COOL, STATE_DRY, STATE_FAN_ONLY, STATE_HEAT,
+    SUPPORT_FAN_MODE, SUPPORT_ON_OFF, SUPPORT_OPERATION_MODE,
+    SUPPORT_SWING_MODE, SUPPORT_TARGET_TEMPERATURE)
+from homeassistant.const import ATTR_TEMPERATURE, CONF_HOST, TEMP_FAHRENHEIT
+import homeassistant.helpers.config_validation as cv
+from homeassistant.util import Throttle
+from homeassistant.util.temperature import convert as convert_temperature
+
+DOMAIN = 'tfiac'
+
+MIN_TIME_BETWEEN_UPDATES = timedelta(seconds=60)
+
+REQUIREMENTS = ['pytfiac==0.3']
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_HOST): cv.string,
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+MIN_TEMP = 61
+MAX_TEMP = 88
+OPERATION_LIST = {
+    STATE_HEAT: 'heat',
+    STATE_AUTO: 'selfFeel',
+    STATE_DRY: 'dehumi',
+    STATE_FAN_ONLY: 'fan',
+    STATE_COOL: 'cool',
+}
+FAN_LIST = ['Auto', 'Low', 'Middle', 'High']
+SWING_LIST = [
+    'Off',
+    'Vertical',
+    'Horizontal',
+    'Both',
+]
+
+CURR_TEMP = 'current_temp'
+TARGET_TEMP = 'target_temp'
+OPERATION_MODE = 'operation'
+FAN_MODE = 'fan_mode'
+SWING_MODE = 'swing_mode'
+ON_MODE = 'is_on'
+
+
+async def async_setup_platform(hass, config, async_add_devices,
+                               discovery_info=None):
+    """Set up the TFIAC climate device."""
+    from pytfiac import Tfiac
+
+    host = config.get(CONF_HOST)
+    if host is not None:
+        tfiac_client = Tfiac(host)
+        await tfiac_client.update()
+        async_add_devices([TfiacClimate(hass, tfiac_client)])
+
+
+class TfiacClimate(ClimateDevice):
+    """TFIAC class."""
+
+    def __init__(self, hass, client):
+        """Init class."""
+        hass.data[DOMAIN] = self
+        self._client = client
+        self._available = True
+
+    @property
+    def available(self):
+        """Return if the device is available."""
+        return self._available
+
+    @Throttle(MIN_TIME_BETWEEN_UPDATES)
+    async def async_update(self):
+        """Update status via socket polling."""
+        try:
+            await self._client.update()
+            self._available = True
+        except futures.TimeoutError:
+            self._available = False
+
+    @property
+    def supported_features(self):
+        """Return the list of supported features."""
+        return (SUPPORT_FAN_MODE | SUPPORT_ON_OFF | SUPPORT_OPERATION_MODE
+                | SUPPORT_SWING_MODE | SUPPORT_TARGET_TEMPERATURE)
+
+    @property
+    def min_temp(self):
+        """Return the minimum temperature."""
+        return convert_temperature(MIN_TEMP, TEMP_FAHRENHEIT,
```

Comment:
We don't need to convert here. Just report in the same unit as `self.temperature_unit`.